### PR TITLE
projects/S905: new project

### PR DESCRIPTION
**Pull request by adamg-xda**
This PR is a new project to support building on devices containing the Amlogic S905 chip.

### CODE REVIEW

File: projects/S905/filesystem/usr/sbin/installtointernal [hunk 1]
```diff
@@ -0,0 +1,116 @@
+#!/bin/sh
+
+################################################################################
+#      This file is part of LibreELEC - https://libreelec.tv
+#      Copyright (C) 2016 kszaq (kszaquitto (at) gmail.com)
```
Comment (CvH) [inline projects/S905/filesystem/usr/sbin/installtointernal]:
```
Copyright (C) 2017-present Team LibreELEC
Copyright (C) 2016 Mateusz Krzak (kszaquitto@gmail.com)
```
I guess ?

File: projects/S905/packages/initramfs/package.mk [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+#  but WITHOUT ANY WARRANTY; without even the implied warranty of
+#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#  GNU General Public License for more details.
+#
+#  You should have received a copy of the GNU General Public License
+#  along with LibreELEC.  If not, see <http://www.gnu.org/licenses/>.
+################################################################################
+
+PKG_NAME="initramfs"
+PKG_VERSION=""
```
Comment (CvH) [inline projects/S905/packages/initramfs/package.mk]: `PKG_VERSION="1.0"` or something

File: projects/S905/packages/initramfs/package.mk [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+PKG_REV="1"
```
Comment (CvH) [inline projects/S905/packages/initramfs/package.mk]: drop `PKG_REV="1"`

File: projects/S905/packages/initramfs/package.mk [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+PKG_ARCH="any"
+PKG_LICENSE="GPL"
+PKG_SITE="https://libreelec.tv"
+PKG_URL=""
+PKG_DEPENDS_TARGET="toolchain libc:init busybox:init linux:init plymouth-lite:init util-linux:init e2fsprogs:init dosfstools:init"
+PKG_SECTION="virtual"
+PKG_SHORTDESC="initramfs: Metapackage for installing initramfs"
+PKG_LONGDESC="debug is a Metapackage for installing initramfs"
+
+PKG_IS_ADDON="no"
```
Comment (CvH) [inline projects/S905/packages/initramfs/package.mk]: drop `PKG_IS_ADDON="no"`

File: projects/S905/packages/initramfs/package.mk (hunk 2 above)
Comment (adamg-xda) [reply projects/S905/packages/initramfs/package.mk]: if you look in packages/virtual/initramfs/package.mk there is also no version specified there

File: projects/S905/packages/initramfs/package.mk (hunk 4 above)
Comment (adamg-xda) [reply projects/S905/packages/initramfs/package.mk]: I think this is a problem with github as this does not exist in my local copy

File: projects/S905/packages/initramfs/package.mk (hunk 3 above)
Comment (adamg-xda) [reply projects/S905/packages/initramfs/package.mk]: As below, doesn't exist in my local copy :confused:

File: projects/S905/packages/initramfs/package.mk (hunk 2 above)
Comment (CvH) [reply projects/S905/packages/initramfs/package.mk]: Its not directly a build problem, if nothing is set you get 0.0-invalid as version. So setting some value is preferred.

File: projects/S905/packages/initramfs/package.mk (hunk 4 above)
Comment (CvH) [reply projects/S905/packages/initramfs/package.mk]: Not sure what happened, as long it is shown here it is likely merged that way. Try to force push your current tree.

File: projects/S905/packages/initramfs/package.mk (hunk 2 above)
Comment (adamg-xda) [reply projects/S905/packages/initramfs/package.mk]: It hasn't been an issue for me before, nevertheless I have changed it

File: projects/S905/packages/initramfs/package.mk (hunk 4 above)
Comment (adamg-xda) [reply projects/S905/packages/initramfs/package.mk]: It's happened before, force pushed a new version and its gone now :)

**Comment by kszaq** [comment]
A few notes if I may:
* `aml-hdmimonitor.sh` is pointless and/or won't work well without a Kodi patch to remove non-CVBS screen resolutions
* there is no `amlogic-3.14-nougat` kernel in master (yet?)
* `installtointernal` should be dropped for now - IIRC we don't want to support installing to internal in official builds - the script can be provided for community as an "addon"
* `rebootfromnand` script may be dropped, at least for now since we don't have patched u-boot-tools in master
* please update storage partition name from `LIBREELEC_DISK` to 'LIBREELEC_STORAGE`, as in https://github.com/LibreELEC/LibreELEC.tv/pull/2260
* you may drop Odroid-C2 specific parts in `bootloaded/update.sh` - from line 93 on
* `/usr/share/bttty.conf` is not supported in firmware package
* please drop custom `advancedsettings.xml`
* please drop project-specific `initramfs` package - it is not needed
* you can drop `odroidc2-pinctrl.patch` as Odroid_C2 is supported in a separate image
* you can drop `vim2-memory.patch` as VIM2 is a S912 device

Overall good job and thank you for taking the burden of submitting this project to master branch.

**Comment by adamg-xda** [comment]
Thankful for your recommendations kszaq I will make the changes later on
tonight

**Comment by kszaq** [comment]
One note: you can continue your work by simply adding commits - rebase, squashing etc can be done at the end.

**Comment by adamg-xda** [comment]
@kszaq I think the update script should be kept as is and moved to maybe aml-dtbtools, so all projects which support multi-dtb can use the same script.

File: packages/linux/package.mk [hunk 5]
```diff
@@ -44,6 +44,13 @@ case "$LINUX" in
     PKG_SOURCE_DIR="$PKG_NAME-amlogic-$PKG_VERSION*"
     PKG_PATCH_DIRS="amlogic-3.14"
     ;;
+  amlogic-3.14-s9xx)
```
Comment (Raybuntu) [inline packages/linux/package.mk]: Please revert that. We don't want more fragmentation for AML, we need more collaboration. All necessary updates should be made to our linux-amlogic fork.

**Comment by adamg-xda** [comment]
All recommended changes pushed

**Pull request closed without merging**